retrofit/src/main/java/retrofit2/Converter.java

```diff
@@ -63,6 +63,17 @@
     }
 
     /**
+     * Returns a {@link Converter} for converting {@code type} to an HTTP request body, or null if
+     * {@code type} cannot be handled by this factory. This is used to create converters for types
+     * specified by {@link Body @Body}, {@link Part @Part}, and {@link PartMap @PartMap}
+     * values.
+     */
+    public Converter<?, RequestBody> requestBodyConverter(Type type,
```

Comment:
While this was made an abstract class so we could add methods to it, I'd prefer to remove the old callback altogether. I don't want to live with both for the next 2-3 years and we're still in beta (the point of which is finding problems with the API like the one you encountered).